jaxtyping/_array_types.py

```diff
@@ -61,30 +61,30 @@ class _DimType(enum.Enum):
     symbolic = enum.auto()
 
 
+@dataclass(frozen=True)
```

Comment:
Actually, IIRC it is important that these *not* compare equal to each other. I don't remember why though.

That said, if that was important we should really have a test about it... maybe let's make this change and see if anyone screams.

Reply:
While I was fixing it I actually was also wondering whether this should be the case, but then I just preferred to have the behavior that is equivalent to the old version. Good point about waiting for screams.